deps/gcc.Dockerfile

```diff
@@ -1,151 +1,39 @@
-ENV CXXFLAGS="$CXXFLAGS -Wno-implicit-fallthrough -Wno-cast-function-type -fpermissive"
-RUN $SRCDIR/$GCC/configure \
-	--prefix=/usr \
-	--target=$TARGET \
-	--enable-languages="c" \
-	--disable-multilib \
-	--with-cpu=$TARGET_CPU \
-	--with-system-zlib \
-	--with-libgloss \
-	--without-headers \
-	--disable-shared \
-	--disable-nls
+RUN apk update && apk upgrade && apk add cmake alpine-sdk wget xz git bash autoconf automake bison flex texinfo help2man gawk libtool ncurses-dev gettext-dev python3-dev rsync openjdk11 ninja
 
-RUN make -j"$(nproc)" all-gcc
-RUN make DESTDIR="$INSTALLDIR/$GCC-bootstrap" install-strip-gcc
-RUN rm -rf "$INSTALLDIR/$GCC-bootstrap/usr/share"
-RUN cp -r "$INSTALLDIR/$GCC-bootstrap/usr/"* /usr/
 #################################################################################
-#############################    Building NewLib    #############################
+############################    Building crosstool-ng    ########################
 #################################################################################
-WORKDIR $BUILDDIR/$NEWLIB
-ENV CFLAGS_FOR_TARGET="-Os -g -ffunction-sections -fdata-sections -fomit-frame-pointer -ffast-math"
+WORKDIR /root
+RUN git clone --branch crosstool-ng-$CROSSTOOL_VERSION --depth=1 https://github.com/crosstool-ng/crosstool-ng.git
+WORKDIR /root/crosstool-ng
+RUN ./bootstrap && ./configure --enable-local
+RUN make
+ENV PATH=/root/crosstool-ng:$PATH
+COPY samples/m68k-elf ./samples/m68k-elf
 
-RUN "$SRCDIR/$NEWLIB/configure" \
-	--prefix=/usr \
-	--target=$TARGET \
-	--enable-languages="c" \
-	--disable-newlib-supplied-syscalls \
-	--disable-multilib \
-	--with-cpu=$TARGET_CPU \
-	--disable-nls
-
-RUN make -j"$(nproc)"
-RUN DESTDIR=$INSTALLDIR/$NEWLIB make install
-RUN cp -r "$INSTALLDIR/$NEWLIB/usr/" "$STAGINGDIR"
-RUN cp -r "$INSTALLDIR/$NEWLIB/usr/"* /usr/
-ENV CFLAGS_FOR_TARGET=""
 #################################################################################
-############################    Building final GCC    ###########################
+###############################    Building gcc   ###############################
 #################################################################################
-WORKDIR $BUILDDIR/$GCC
-
-RUN "$SRCDIR/$GCC/configure" \
-	--prefix=/usr \
-	--target=$TARGET \
-	--enable-languages="c,c++" \
-	--disable-multilib \
-	--with-cpu=$TARGET_CPU \
-	--with-system-zlib \
-	--with-newlib \
-	--with-libgloss \
-	--disable-shared \
-	--disable-nls
-
-RUN make -j"$(nproc)"
-RUN make DESTDIR="$INSTALLDIR/$GCC" install-strip
-RUN rm -rf "$INSTALLDIR/$GCC/usr/share"
-RUN rm "$INSTALLDIR/$GCC/usr/lib/libcc1.so"
-RUN rm "$INSTALLDIR/$GCC/usr/lib/libcc1.so.0"
-RUN rm "$INSTALLDIR/$GCC/usr/lib/libcc1.so.0.0.0"
-
-RUN cp -ar "$INSTALLDIR/$GCC/usr/" "$STAGINGDIR"
-
+WORKDIR $BASEDIR
+ENV CT_EXPERIMENTAL=y
+ENV CT_ALLOW_BUILD_AS_ROOT=y
+ENV CT_ALLOW_BUILD_AS_ROOT_SURE=y
+# Configure and build the tools
+RUN ct-ng m68k-elf
```

Comment:
Configure and build the tools.

Reply:
not sure what GCC 13.2 brings for m68j compared to GCC 6.3, from my tests later version perform worse on code generation but I admit I only tested up to GCC 10.
Would be nice to compare 'benchmark' sample compiled with GCC 6.3 versus GCC 13.2 to see the difference.
My way of seeing things is always : the simpler and the lighter, the better...
I'm almost certain the GCC 13.2 toolchain is heavier than GCC 6.x but if that comes with some benefits why not.
Also i'm not sure about CMake requirement, again, what does it brings here ?

Reply:
@Stephane-D , In general I tend to agree. I did run the benchmark sample and there are some improvements and a few things are worse. GCC 13.2.0 gives a slightly better overall score of 15247 compared to 14298 from GCC 6.3. These results are from blastem on Windows. I need to try on real hardware.

I think of CMake as a build description language. You describe how to build your project and CMake handles the generation of the actual build scripts, including handling all the dependancies. It supports generating build scripts for a number of standard tools like Make and Ninja. I find Ninja to be faster than using Make in many cases, especially in msys or mingw environments. It's also good at keeping the derived files contained so you can do a debug or a release build without having to perform a clean inbetween, etc.

Let me know what you think. I've you don't want to move to gcc 13.2 I can change the config to build 6.5 which is closer to what you have now.

Reply:
The better performance on GCC 13.2 is a good surprise ! So definitely you can keep it.
About CMake I understand it interest if it helps for the Docker usage, so it's all good for me, thanks :)